tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -184,6 +186,61 @@ def check_sr_subjects():
 
         wait_until(check_sr_subjects, timeout_sec=5)
 
+    @cluster(num_nodes=4)
+    def test_java_client(self):
+        kc_node = self.keycloak.nodes[0]
+
+        self.keycloak.admin.create_user('norma',
+                                        'desmond',
+                                        realm_admin=True,
+                                        email='10086@sunset.blvd')
+        self.keycloak.login_admin_user(kc_node, 'norma', 'desmond')
+        self.keycloak.admin.create_client(CLIENT_ID)
+
+        # add an email address to myapp client's service user. this should
+        # appear alongside the access token.
+        self.keycloak.admin.update_user(f'service-account-{CLIENT_ID}',
+                                        email='myapp@customer.com')
+
+        self.rpk.create_topic(EXAMPLE_TOPIC)
+        service_user_id = self.keycloak.admin_ll.get_user_id(
+            f'service-account-{CLIENT_ID}')
+        result = self.rpk.sasl_allow_principal(f'User:{service_user_id}',
+                                               ['all'], 'topic', EXAMPLE_TOPIC,
+                                               self.su_username,
+                                               self.su_password,
+                                               self.su_algorithm)
+
+        cfg = self.keycloak.generate_oauth_config(kc_node, CLIENT_ID)
+        assert cfg.client_secret is not None
+        assert cfg.token_endpoint is not None
+
+        cli = KafkaCliTools(self.redpanda,
+                            user=self.su_username,
+                            passwd=self.su_password,
```

Comment:
Is it possible to not specify these superuser credentials?

Reply:
You mean to use OAuth instead?

Reply:
> You mean to use OAuth instead?

Yes

Reply:
Skipped it because I thought it was gonna be painful; then realized it wouldn't be, so I've done it. It was painful.